### PR TITLE
Remove six-related code and __future__ imports

### PR DESCRIPTION
**Pull request by elacuesta**
Based on #4115 and #4114, supersedes #4003

### CODE REVIEW

**Comment by elacuesta** [comment]
@Gallaecio I agree with the proposed sorting suggestions, but I believe the PR is already large enough as it is (that said, I probably already sorted some of them). If you don't mind, I'd prefer to leave that import cleanup for separate PRs. What do you think?

File: scrapy/http/headers.py [hunk 1]
```diff
@@ -68,9 +67,6 @@ def appendlist(self, key, value):
         self[key] = lst
 
     def items(self):
-        return list(self.iteritems())
```
[the remark this answers is not in the document]
Comment (elacuesta) [reply scrapy/http/headers.py]: I believe returning a generator is more consistent with the py3 behaviour. Ideally we should return a `dict_items` object, but I don't think there is a way to create them without creating a `dict` first, which is probably not very performant.

**Comment by elacuesta** [comment]
Added two pickle-related checks to the bandit skip list (B301, B403). The problem appeared with the change from `from six.moves import cPickle as pickle` to `import pickle`. I'm not sure what else to do besides skipping the check, I thought #4135 might make a difference but I'm not sure, since B403 fails early at the import stage.

**Comment by Gallaecio** [comment]
Pickle is always a potential danger if it handles uncontrolled input data, I believe that’s why bandit checks it. If we are sure that a given pickle usage is safe, I think we can just mark that specific usage as such.

**Pull request merged**